tools/module_dependencies.py

```diff
@@ -0,0 +1,82 @@
+__all__ = ['modules_dependent_on', 'dependency_graph']
+
+
+import numpy as np
+import importlib
+import multiprocessing
+import functools
+import sys
+
+
+package = 'skimage'
+included_mods = [
+    '_shared',
+]
+excluded_mods = [
+    '__version__',
+]
+
+
+@functools.cache
+def _pkg_modules() -> list[str]:
+    pkg = importlib.import_module(package)
+
+    members = {f'{package}.{attr}' for attr in dir(pkg)}
+    included_mods_full = {f'{package}.{mod}' for mod in included_mods}
+    excluded_mods_full = {f'{package}.{mod}' for mod in excluded_mods}
+
+    return (members | set(included_mods_full)) - set(excluded_mods_full)
+
+
+@functools.cache
+def _pkg_modules_index():
+    mods = sorted(_pkg_modules())
+    return {mod: index for index, mod in enumerate(mods)}
+
+
+def _import_dependencies(module: str | list[str]) -> set[str]:
+    importlib.import_module(module)
+    importlib.import_module(f"{module}.tests")
+
+    pkg_modules = _pkg_modules()
+    pkg_sys_modules = {mod for mod in sys.modules if package in mod}
+
+    return pkg_modules & pkg_sys_modules
+
+
+def dependency_graph():
+    """Calculate the dependency graph of the current module.
+
+    Each row represents the dependencies of one subpackage.
+
+    """
+    mods = sorted(_pkg_modules())  # sort for stable matrix
```

Comment:
This is sorted in three places but used as a set only in `_import_dependencies`. Might be simpler to do the sorting in `_pkg_modules` and convert back into a set if needed only in `_import_dependencies`. I don't think this will be noticeable in performance either way; this is just for simplicity.

Reply:
Yeah, the tradeoff in performance vs simplicity. Of course, it could return both as a compromise. I guess I value the simplicity (to get something adopted) more than the performance.